src/3d/Target.fst

```diff
@@ -82,17 +82,37 @@ let namespace_of_integer_type =
    | UInt32 -> "UInt32"
    | UInt64 -> "UInt64"
 
-let print_op =
-  function
+let print_range (r:A.range) : string =
+  let open A in
+  Printf.sprintf "(Prims.mk_range \"%s\" %d %d %d %d)"
+    (fst r).filename
+    (fst r).line
+    (fst r).col
+    (snd r).line
+    (snd r).col
+
+let print_arith_op
+  (o:op{Plus? o \/ Minus? o \/ Mul? o \/ Division? o})
+  (t:A.integer_type)
+  (r:option A.range)
+  = let t = match t with | A.UInt8 -> "u8" | A.UInt16 -> "u16" | A.UInt32 -> "u32" | A.UInt64 -> "u64" in
+    let fn = match o with | Plus _ -> "add" | Minus _ -> "sub" | Mul _ -> "mul" | Division _ -> "div" in
+    let r = match r with | Some r -> r | None -> A.dummy_range in
+
+    Printf.sprintf "`Prelude.%s_%s \"Cannot verify that %s %s is safe\" %s`"

```

Comment:
The "Cannot verify that ... is safe" error messages look quite general. Thus, we could add them once and for all in the definitions of `Prelude.u32_mul` etc. instead of generating them in the .Types.fst

Reply:
Done